spmd/tensor/device_mesh.py

```diff
@@ -426,16 +457,23 @@ def reduce_scatter(
         ), "Rank if not part of mesh"  # TODO: figure out behavior here
 
         num_chunks = self.size(mesh_dim)
-        if self.backend() == "nccl":
+        if self._backend == "nccl":
             # gathered_list = output_tensor.tensor_split(num_chunks, dim=tensor_dim)
             input_list = list(input.tensor_split(num_chunks, dim=tensor_dim))
             output = torch.empty_like(input_list[my_coordinate])
             dim_group = self._dim_groups[mesh_dim]
             reduce_scatter(output, input_list, op=op, group=dim_group)
             return output
-        else:
+        elif self._backend == "gloo":
             # it's gloo, which does not have reduce_scatter
             # we have to do all_reduce + scatter
+            # warnings.warn(
```

Comment:
Is there a reason this warning is commented out?

Reply:
just added this to the PR above which handles reduce_scatter, agree that it's valuable to warn it explicitly :)